### ISSUE TITLE
Enum values that are defined in OpenAPI spec don't appear to be set properly in generated types

### ISSUE THREAD

**Issue by robin24** [apple/swift-openapi-generator#454]
As part of my OpenAPI spec, I have an enum mapping continents to their abbreviations. In the spec this looks as follows:
```
    ContinentEnum:
      enum:
      - AF
      - AN
      - AS
      - EU
      - NA
      - OC
      - SA
      type: string
      description: |-
        * `AF` - Africa
        * `AN` - Antarctica
        * `AS` - Asia
        * `EU` - Europe
        * `NA` - North America
        * `OC` - Oceania
        * `SA` - South America
```
The openapi-generator code generated for that definition looks as follows:
```
        /// * `AF` - Africa
        /// * `AN` - Antarctica
        /// * `AS` - Asia
        /// * `EU` - Europe
        /// * `NA` - North America
        /// * `OC` - Oceania
        /// * `SA` - South America
        ///
        /// - Remark: Generated from `#/components/schemas/ContinentEnum`.
        @frozen internal enum ContinentEnum: String, Codable, Hashable, Sendable {
            case AF = "AF"
            case AN = "AN"
            case AS = "AS"
            case EU = "EU"
            case NA = "NA"
            case OC = "OC"
            case SA = "SA"
        }
```
It thus appears that instead of using the values provided in the spec for each enum case, the generator assigns the key of each case as also being the value.
Given the above, I'd personally prefer to see these enum cases defined with the values from the OpenApi document, e.g.
```
            case AF = "Africa"
            case AN = "Antarctica"
            case AS = "Asia"
            case EU = `"Europe"
```
I'm also not sure if this behavior is expected, or perhaps an issue with my OpenAPI spec.

**Comment by czechboy0** [comment]
The OpenAPI specification doesn't assign any special meaning to `description`, it's just freeform text. There's no place in the OpenAPI document to provide the "name" for a case that's distinct from its raw value.

So unfortunately this is a limitation of the OpenAPI 3.1.0 specification itself, not of this generator.

---

Now, technically, this generator _could_ try to somehow parse patterns from the description, but since those patterns are completely freeform, it's unlikely to work reliably, and would result in the generator having to support various heuristics that aren't required by the OpenAPI specification.

---

To summarize, your OpenAPI document is correct, and is the best you can do with OpenAPI 3.1.0. If this is an issue for you, you could try engaging with the OpenAPI specification community and pitch an improvement for a future revision of the OpenAPI specification.

**Comment by robin24** [comment]
Thank you so much for the clarification, it's super greatly appreciated and really helped me to better understand why this is how it is.

Surprised to hear there's no official specification for that as enums are quite common IMO. Not a huge deal to me personally as I can just solve this by rolling a wrapper of my own to have the proper values assigned, but I sure will look at their process for making proposals when I find the time.
Thanks again for the wonderful help!

**Comment by czechboy0** [comment]
You're welcome!

One (possibly not very pretty) alternative idea would be to use a oneOf of string enums, each with one case.

So instead of:

```yaml
type: string
enum:
  - A
  - B
  - C
```

You could use:

```yaml
oneOf:
  - type: string
    description: My custom A description
    enum: [A]
  - type: string
    description: My custom B description
    enum: [B]
  - type: string
    description: My custom C description
    enum: [C]
```

Although it'll generate a bit more verbose code, and will likely be less performant. But just thought I'd mention it.

**Comment by robin24** [comment]
Ohhh, that does seem like a super smart suggestion considering the current limitations with enums, thanks for putting that in my head! :-)
Since I developed the server for this in Python using the Django framework, I'm using [drf-spectacular](https://github.com/tfranzel/drf-spectacular) for OpenAPI schema generation.
It's a quite modular package, so I'll just play with this a little and see if I can hack something together that changes how these enums are defined in the spec file. 👍

**Comment by robin24** [comment]
Ah, gotcha. It was looking alright from the example you provided, but I absolutely agree that having value1, value2 etc as case names wouldn't be desirable, and in fact would be harder to reason about than what we have right now.

Another thought, is this something which could be addressed by a generator plugin at a future point, similar to what is described in #112?
Not necessarily saying a plugin for this should be provided as part of this package, but maybe a way for adopters to override certain schema generation logic if desired.

**Comment by czechboy0** [comment]
For sure open to ideas for other plugin types. If you have a specific idea of the function kind you'd want to implement in a plugin (e.g., for the example in #112 it's something like `func swiftSafeName(name: String, kind: ...) -> String`), please add a comment to #112.
